lib/Xmds/Soap.php

```diff
@@ -1552,21 +1536,15 @@ protected function doSubmitStats($serverKey, $hardwareKey, $statXml)
                 } catch (NotFoundException $notFoundException) {
                     // Widget isn't found
                     // we can only log this and move on
-                    $this->getLog()->info('Stat for a widgetId that doesnt exist: ' . $widgetId);
+                    $this->getLog()->error('Stat for a widgetId that doesnt exist: ' . $widgetId);
                     continue;
                 }
             }
-            
-            $tag = $node->getAttribute('tag');
 
+            $tag = $node->getAttribute('tag');
             if ($tag == 'null')
                 $tag = null;
 
-            if ($fromdt > $todt) {
```

Comment:
Seems like this logic has been lost during the changes? It makes sense to have this happen when we are checking the other date/time bits above?